FWCore/ParameterSet/python/Modules.py

```diff
@@ -551,6 +569,19 @@ def testSwitchProducer(self):
             self.assertRaises(TypeError, lambda: SwitchProducerTest(test1 = ESPrefer("Foo")))
             self.assertRaises(TypeError, lambda: SwitchProducerTest(test1 = SwitchProducerTest(test1 = EDProducer("Foo"))))
 
+            # Label
+            sp.setLabel("sp")
+            self.assertEqual(sp.label_(), "sp")
+            self.assertEqual(sp.test1.label_(), "sp@test1")
+            self.assertEqual(sp.test2.label_(), "sp@test2")
+            sp.test3 = EDProducer("Xyzzy")
+            self.assertEqual(sp.test3.label_(), "sp@test3")
+            sp.test1 = EDProducer("Fred")
+            self.assertEqual(sp.test1.label_(), "sp@test1")
+            del sp.test1
+            sp.test1 = EDProducer("Wilma")
+            self.assertEqual(sp.test1.label_(), "sp@test1")
+
```

Comment:
Do you want to test what happens if you call `setLabel` again with a different value?

Reply:
The latter is what I was thinking.